versions-common/src/main/java/org/codehaus/mojo/versions/ordering/BoundArtifactVersion.java

```diff
@@ -1,44 +1,106 @@
 package org.codehaus.mojo.versions.ordering;
 
-import java.util.Iterator;
+import java.util.ArrayList;
 import java.util.List;
 
 import org.apache.commons.lang3.builder.EqualsBuilder;
 import org.apache.commons.lang3.builder.HashCodeBuilder;
 import org.apache.maven.artifact.versioning.ArtifactVersion;
-import org.apache.maven.artifact.versioning.DefaultArtifactVersion;
 import org.codehaus.mojo.versions.api.Segment;
-
-import static org.codehaus.mojo.versions.ordering.ComparableVersion.IntegerItem.ZERO;
+import org.codehaus.mojo.versions.utils.DefaultArtifactVersionCache;
+import org.codehaus.plexus.util.StringUtils;
 
 /**
- * <p>Represents an artifact version with all segments more major or equal to a given segment
- * held in place. It can be thought of as an artifact having +&infin; as its upper bound
- * on all segments less major than the held segment.</p>
- * <p>When compared with another artifact versions, this results with the other object
+ * <p>Represents an <b>immutable</b> artifact version with all segments <em>major</em> to the given segment
+ * held in place. It can be thought of as an artifact having +∞ as its upper bound
+ * on all segments minor to the held segment.</p>
+ * <p>For example:</p>
+ * <p>A {@link BoundArtifactVersion} of {@code [1.2.3-2, INCREMENTAL]} can be seen as {@code 1.2.+∞}
+ * and will be greater than all versions matching the {@code 1.2.*} pattern.</p>
+ * <p>A {@link BoundArtifactVersion} of {@code [1.2.3-2, SUBINCREMENTAL]} will be greater
+ *  * than all versions matching the {@code 1.2.3-2.*} pattern.</p>
+ * <p>When compared to another artifact versions, this results with the other object
  * with the segment versions up to the held segment being equal,
  * always comparing lower than this object.</p>
  * <p>This is particularly helpful for -SNAPSHOT and other versions with qualifiers, which
  * are lower than version 0 in the Maven versioning system.</p>
  */
-public class BoundArtifactVersion extends DefaultArtifactVersion {
+public class BoundArtifactVersion implements ArtifactVersion {
     /**
      * Most major segment that can change, i.e. not held in place.
      * All segments that are more major than this one are held in place.
      */
     private final Segment segment;
 
-    private final BoundComparableVersion comparator;
+    private final ArtifactVersion comparable;
 
     /**
-     * Constructs the instance
+     * Constructs the instance given the version in a text format.
+     * @param artifactVersion version in a text format
+     * @param segment most major segment that can change, i.e. <em>not</em> held in place
+     */
+    public BoundArtifactVersion(String artifactVersion, Segment segment) {
```

Comment:
So this is the key method, I guess... 

Hope this is clear, looks a bit like first year's comp-sci exercise, but we're simply parsing the version and splitting it to tokens upon:
- a change from numbers to letters or vice-versa
- an - character
- a dot

The new constructed version will have Inetger.MAX_VALUE on all places less than the fixed segment. Also, all non-strandard tokens will be replaced by zeros (like in the old impl).